crypto/sha/asm/sha512-x86_64.pl

```diff
@@ -2313,6 +2383,248 @@ ()
+
+$code.=<<___;
+	or	$arg_num_blks, $arg_num_blks
+	je	.Lsha512ext_done
+___
+
+$code.=<<___ if($win64);
+	# xmm6:xmm9, xmm11:xmm15 need to be maintained for Windows
+	# xmm10 not used
+	sub	\$`9*16`,%rsp
+.cfi_adjust_cfa_offset	\$`9*16`
+	vmovdqu	%xmm6, 16*0(%rsp)
+	vmovdqu	%xmm7, 16*1(%rsp)
+	vmovdqu	%xmm8, 16*2(%rsp)
+	vmovdqu	%xmm9, 16*3(%rsp)
+	vmovdqu	%xmm11,16*4(%rsp)
+	vmovdqu	%xmm12,16*5(%rsp)
+	vmovdqu	%xmm13,16*6(%rsp)
+	vmovdqu	%xmm14,16*7(%rsp)
+	vmovdqu	%xmm15,16*8(%rsp)
+___
+
+$code.=<<___;
+	# reuse shuffle indices from $K512
+	vbroadcasti128	0x500+$TABLE(%rip),%ymm15
+
+	# load current hash value and transform
+	vmovdqu	0x00($arg_hash),%ymm0
+	vmovdqu	0x20($arg_hash),%ymm1
+	# ymm0 = DCBA, ymm1 = HGFE
+	vperm2i128	\$0x20,%ymm1,%ymm0,%ymm2
+	vperm2i128	\$0x31,%ymm1,%ymm0,%ymm3
+	# ymm2 = FEBA, ymm3 = HGDC
+	vpermq	\$0x1b,%ymm2,%ymm13
+	vpermq	\$0x1b,%ymm3,%ymm14
+	# ymm13 = ABEF, ymm14 = CDGH
+
+	lea	${TABLE}_single(%rip),%r9
+
+.align 32
+.Lsha512ext_block_loop:
+
+	vmovdqa	%ymm13,%ymm11					# ABEF
+	vmovdqa	%ymm14,%ymm12					# CDGH
+
+	# R0 - R3
+	vmovdqu	0*32($arg_msg),%ymm0
+	vpshufb	%ymm15,%ymm0,%ymm3				# ymm0/ymm3 = W[0..3]
+	vpaddq	0*32(%r9),%ymm3,%ymm0
+	vsha512rnds2	%xmm0,%ymm11,%ymm12
+	vperm2i128	\$0x1,%ymm0,%ymm0,%ymm0
+	vsha512rnds2	%xmm0,%ymm12,%ymm11
+
+	# R4 - R7
+	vmovdqu	1*32($arg_msg),%ymm0
+	vpshufb	%ymm15,%ymm0,%ymm4				# ymm0/ymm4 = W[4..7]
+	vpaddq	1*32(%r9),%ymm4,%ymm0
+	vsha512rnds2	%xmm0,%ymm11,%ymm12
+	vperm2i128	\$0x1,%ymm0,%ymm0,%ymm0
+	vsha512rnds2	%xmm0,%ymm12,%ymm11
+	vsha512msg1	%xmm4,%ymm3					# ymm3 = W[0..3] + S0(W[1..4])
+
+	# R8 - R11
+	vmovdqu	2*32($arg_msg),%ymm0
+	vpshufb	%ymm15,%ymm0,%ymm5				# ymm0/ymm5 = W[8..11]
+	vpaddq	2*32(%r9),%ymm5,%ymm0
+	vsha512rnds2	%xmm0,%ymm11,%ymm12
+	vperm2i128	\$0x1,%ymm0,%ymm0,%ymm0
+	vsha512rnds2	%xmm0,%ymm12,%ymm11
+	vsha512msg1	%xmm5,%ymm4					# ymm4 = W[4..7] + S0(W[5..8])
+
+	# R12 - R15
+	vmovdqu	3*32($arg_msg),%ymm0
+	vpshufb	%ymm15,%ymm0,%ymm6				# ymm0/ymm6 = W[12..15]
+	vpaddq	3*32(%r9),%ymm6,%ymm0
+	vpermq	\$0x1b,%ymm6,%ymm8				# ymm8 = W[12] W[13] W[14] W[15]
+	vpermq	\$0x39,%ymm5,%ymm9				# ymm9 = W[8]  W[11] W[10] W[9]
+	vpblendd	\$0x3f,%ymm9,%ymm8,%ymm8	# ymm8 = W[12] W[11] W[10] W[9]
+	vpaddq	%ymm8,%ymm3,%ymm3				# ymm3 = W[0..3] + S0(W[1..4]) + W[9..12]
+	vsha512msg2	%ymm6,%ymm3					# W[16..19] = ymm3 + S1(W[14..17])
+	vsha512rnds2	%xmm0,%ymm11,%ymm12
+	vperm2i128	\$0x1,%ymm0,%ymm0,%ymm0
+	vsha512rnds2	%xmm0,%ymm12,%ymm11
+	vsha512msg1	%xmm6,%ymm5					# ymm5 = W[8..11] + S0(W[9..12])
+___
+
+# R16 - R63
+for($i=4;$i<16;) {
+$code.=<<___;
+	# R16 - R19, R32 - R35, R48 - R51
+	vpaddq	`$i * 32`(%r9),%ymm3,%ymm0
+	vpermq	\$0x1b,%ymm3,%ymm8				# ymm8 = W[16] W[17] W[18] W[19]
+	vpermq	\$0x39,%ymm6,%ymm9				# ymm9 = W[12] W[15] W[14] W[13]
+	vpblendd	\$0x3f,%ymm9,%ymm8,%ymm7	# ymm7 = W[16] W[15] W[14] W[13]
+	vpaddq	%ymm7,%ymm4,%ymm4				# ymm4 = W[4..7] + S0(W[5..8]) + W[13..16]
+	vsha512msg2	%ymm3,%ymm4					# W[20..23] = ymm4 + S1(W[18..21])
```

Comment:
indentation of comments looks bit off here and below (above too)

Reply:
GitHub using 8 spaces per tab might cause it.
Replaced tabs by 4 spaces in new code according to [coding style](https://openssl-library.org/policies/technical/coding-style/).